kube_aws_autoscaler/main.py

```diff
@@ -190,7 +190,7 @@ def slow_down_downscale(asg_sizes: dict, nodes_by_asg_zone: dict):
     for asg_name, desired_size in sorted(asg_sizes.items()):
         amount_of_downscale = node_counts_by_asg[asg_name] - desired_size
         if amount_of_downscale >= 2:
-            new_desired_size = node_counts_by_asg[asg_name] - 1
+            new_desired_size = max(desired_size * 0.95, node_counts_by_asg[asg_name] * 0.95)
```

Comment:
https://stackoverflow.com/questions/31818050/python-2-7-round-number-to-nearest-integer
* You must round and cast to int
* The logic is incorrect, your ultimate goal is to downscale to desired_size by no more than by -5%at a time, that is
import math
new_desired_size=max(desired_size, int(math.floor(0.95*node_counts_by_asg[asg_name])))